src/GrumPHP/Locator/ChangedFiles.php

```diff
@@ -41,8 +42,20 @@ public function __construct(Repository $repository, Filesystem $filesystem)
     public function locateFromGitRepository()
     {
         $diff = $this->repository->getWorkingCopy()->getDiffStaged();
+//      if($context instanceof GitPrePushContext ) {
+        $actualbranch = trim(shell_exec('git name-rev --name-only HEAD'));
```

Comment:
Try to avoid shell_exec. This is untestable code.
You can use the gitonomy library to run git commands through the ProcessBuilder:
`\Gitonomy\Git\Repository::run()`